src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -330,6 +330,9 @@ enum IntrinsicId
             Expression_New,
             Enum_GetValues,
             Marshal_SizeOf,
+            Marshal_PtrToStructure,
+            Marshal_DestroyStructure,
+            Marshal_GetDelegateForFunctionPointer,
```

Comment:
`Marshal.GetFunctionPointerForDelegate` as well?

Reply:
Ah. Didn't see it in Intellisense and apparently we obsoleted and editorbrowsable(never)-hidden it.

Seems like the other non-generic version is obsolete. But people still use it. Doesn't hurt to add support for it.

Reply:
I think it has the same problem as `StructureToPtr`. It takes object, not the type. So we can just emit warning for it.